File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/rust_solutions/665-non-decreasing-array.rs
impl Solution {
    pub fn check_possibility(nums: Vec<i32>) -> bool {
        let mut count = 0;
        let mut nums = nums;
        for i in 1..nums.len() {
            if nums[i] < nums[i - 1] {
                count += 1;
                if count > 1 {
                    return false;
                }
                if i > 1 && nums[i] < nums[i - 2] {
                    nums[i] = nums[i - 1];
                } else {
                    nums[i - 1] = nums[i];
                }
            }
        }
        true
    }
}

File: leetcode/AIcoding/5.Qwen-3/javascript_solutions/665-non-decreasing-array.js
function checkPossibility(nums) {
    let count = 0;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] < nums[i - 1]) {
            if (count > 0) return false;
            if (i === 1 || nums[i] >= nums[i - 2]) {
                nums[i - 1] = nums[i];
            } else {
                nums[i] = nums[i - 1];
            }
            count++;
        }
    }
    return true;
}